### PR TITLE
Add VolumeMounts to Prometheus

### PR DESCRIPTION
**Pull request by paulfantom**
Same as #2755 but for prometheus.

/cc @LiliC @s-urbaniak

### CODE REVIEW

**Comment by s-urbaniak** [comment]
LGTM

**Comment by s-urbaniak** [comment]
@LiliC Agreed, but I also recall a discussion with @brancz that strategic merge patch is a workaround'ish solution. For consistency I still advocate getting this in, wdyt?

**Comment by paulfantom** [comment]
@LiliC wasn't this something you wanted to do as a follow-up to #2755?

File: pkg/prometheus/statefulset.go [hunk 1]
```diff
@@ -552,6 +552,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		},
 	}
 
+	promVolumeMounts = append(promVolumeMounts, p.Spec.VolumeMounts...)
```
Comment (LiliC) [inline pkg/prometheus/statefulset.go]: I am guessing this won't interfere with the merge patch we do?

**Comment by LiliC** [comment]
@paulfantom so we had an offline discussion on this.

This PR to add support of volumemounts can be merged. After that we should open an issue and also add merge patch for Alertmanager resource so we have consistency.

**Comment by s-urbaniak** [comment]
@paulfantom ping, can you rebase. We have consensus this to be merged.

**Comment by paulfantom** [comment]
rebased

**Comment by lilic** [comment]
I think we are also missing this:
> After that we should open an issue and also add merge patch for Alertmanager resource so we have consistency.

Otherwise this lgtm

**Comment by s-urbaniak** [comment]
@lilic @paulfantom i was under assumption that merge patch is already available for alertmanager, no?

File: pkg/prometheus/statefulset.go (hunk 1 above)
Comment (pgier) [reply pkg/prometheus/statefulset.go]: It should be fine because the container merge patch happens later.

**Comment by pgier** [comment]
> After that we should open an issue and also add merge patch for Alertmanager resource so we have consistency.

Created #3080 to address this.

**Comment by s-urbaniak** [comment]
@paulfantom sorry for sounding like a broken record, but this needs another rebase :disappointed:

**Comment by paulfantom** [comment]
rebased

**Comment by satishmane** [comment]
Hi @paulfantom 
Adding volumeMounts to prometheus container is available in which release version

**Pull request merged**